src/main/clojure/aerospike_clj/client.clj

```diff
@@ -172,390 +115,359 @@
-   (put-multiple db indices set-names payloads expirations {}))
-  ([db indices set-names payloads expirations conf]
-   (p/all
-     (map (fn [[index set-name payload expiration]]
-            (put db index set-name payload expiration conf))
-          (map vector indices set-names payloads expirations)))))
-
-(defn set-single
-  "`put` with an update policy"
-  ([db index set-name data expiration]
-   (set-single db index set-name data expiration {}))
-  ([db index set-name data expiration conf]
-   (_put db
-         index
-         ((:transcoder conf identity) data)
-         (policy/set-policy (get-client db) expiration)
-         set-name)))
-
-(defn create
-  "`put` with a create-only policy"
-  ([db index set-name data expiration]
-   (create db index set-name data expiration {}))
-  ([db index set-name data expiration conf]
-   (_put db
-         index
-         ((:transcoder conf identity) data)
-         (policy/create-only-policy (get-client db) expiration)
-         set-name)))
-
-(defn replace-only
-  "`put` with a replace-only policy"
-  ([db index set-name data expiration]
-   (replace-only db index set-name data expiration {}))
-  ([db index set-name data expiration conf]
-   (_put db
-         index
-         ((:transcoder conf identity) data)
-         (policy/replace-only-policy (get-client db) expiration)
-         set-name)))
-
-(defn update
-  "Writing a new value for the key `index`.
-  Generation: the expected modification count of the record (i.e. how many times was it
-  modified before my current action). Pass a function in `(:trascoder conf)` to modify
-  `data` before it is sent to the DB."
-  ([db index set-name new-record generation new-expiration]
-   (update db index set-name new-record generation new-expiration {}))
-  ([db index set-name new-record generation new-expiration conf]
-   (_put db
-         index
-         ((:transcoder conf identity) new-record)
-         (policy/update-policy (get-client db) generation new-expiration)
-         set-name)))
-
-(defn add-bins
-  "Add bins to an existing record without modifying old data. The `new-data` must be a
-  Clojure map."
-  ([db index set-name ^IPersistentMap new-data new-expiration]
-   (add-bins db index set-name new-data new-expiration {}))
-  ([db index set-name ^IPersistentMap new-data new-expiration conf]
-   (_put db
-         index
-         ((:transcoder conf identity) new-data)
-         (policy/update-only-policy (get-client db) new-expiration)
-         set-name)))
-
-(defn touch
-  "Updates the ttl of the record stored under at `index` to `expiration` seconds from now.
-  Expects records to exist."
-  [db index set-name expiration]
-  (let [client (get-client db index)
-        op-future (p/deferred)
-        start-time (System/nanoTime)]
-    (.touch ^AerospikeClient client
-            ^EventLoop (.next ^EventLoops (:el db))
+    (register-events op-future client-events :write index start-time)))
+
+(defrecord SimpleAerospikeClient [client
```

Comment:
maybe a `deftype`?

Reply:
Done. Had to align initializing instances (did you know that `deftype`s don't have a `map->*` constructor?) and tests as the instance lost its map interface and its class members have to be accessed with `.-*`.